pkg/build/api/types.go

```diff
@@ -397,6 +397,9 @@ const (
 	// availability of a new version of an image
 	ImageChangeBuildTriggerType           BuildTriggerType = "ImageChange"
 	ImageChangeBuildTriggerTypeDeprecated BuildTriggerType = "imageChange"
+
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	ConfigChangeBuildTriggerType BuildTriggerType = "ConfigChange"
```

Comment:
Good question, yes the intent is to eventually do a full config change controller. BuildConfigs would get the new behavior when that change comes. @bparees @smarterclayton thoughts?